CDCgov/prime-reportstream
# REST Transport - Issuer and Audience Customization

This PR introduces the capability to use a map in the settings file to generate JWTs for REST transport. The implementation focuses on retrieving values under the `iss` (issuer) and `aud` (audience) keys. This enhancement allows for the creation of JWTs with custom issuer and audience fields, applicable specifically to the `"two-legged"` `authType`.

This feature is necessary to establish a connection with EPIC - UCSD using `OAuth2`. Although TI is the issuer of the token, we need to utilize the client ID provided by EPIC to create the JWT. With these settings, we can modify the issuer from TI-sender to EPIC's client ID and set the audience to EPIC's token endpoint.


Test Steps:
1. Run `gradlew :prime-router:clean package`
2. Run `gradlew :prime-router:testIntegration`
3. Create receiver with REST transport with `authType: "two-legged`
4. Add `jwtParams:` under `transport:`
5. Stub `"iss":""` and `"aud":""` with mock entries, under `jwtParams` ** see [Reference section](#reference) **
7. Fire up the containers
8. Add the new receiver 
9. Send file to api
10. Make sure jwt gets created using the `iss` and `aud` from the `jwtParams` map


## Changes
- `TransportType.kt`
  - Added `jwtParams` map as part of constructor
- `RESTTransport.kt`
  - Added functionality to use the `jwtParams` map to generate jwt
- `RESTTransportIntegrationTests.kt`
  - Added integration test to account when using `jwtParams` to create jwt
- `gitleaks-config.toml`
  - Added regex to allow line of code to be excluded from gitleaks

## Checklist

### Testing

<a name="reference"></a>
## Reference
transport settings using `jwtParams`
![image](https://github.com/CDCgov/prime-reportstream/assets/49923512/cffebb3a-af03-453a-ac11-8dc8d87f680d)
